airbyte-server/src/main/java/io/airbyte/server/scheduler/DefaultSynchronousSchedulerClient.java

```diff
@@ -101,15 +106,19 @@ public SynchronousResponse<StandardCheckConnectionOutput> createDestinationCheck
   }
 
   @Override
-  public SynchronousResponse<AirbyteCatalog> createDiscoverSchemaJob(final SourceConnection source, final String dockerImage)
+  public SynchronousResponse<String> createDiscoverSchemaJob(final SourceConnection source, final String dockerImage, final String connectorVersion)
       throws IOException {
     final JsonNode sourceConfiguration = oAuthConfigSupplier.injectSourceOAuthParameters(
         source.getSourceDefinitionId(),
         source.getWorkspaceId(),
         source.getConfiguration());
     final JobDiscoverCatalogConfig jobDiscoverCatalogConfig = new JobDiscoverCatalogConfig()
         .withConnectionConfiguration(sourceConfiguration)
-        .withDockerImage(dockerImage);
+        .withDockerImage(dockerImage)
+        .withSourceId(source.getSourceId().toString())
+        .withConfigHash(HASH_FUNCTION.hashBytes(Jsons.serialize(source.getConfiguration()).getBytes(
```

Comment:
This feels like the kind of thing that should maybe go into a helper or util class, and we can add some javadoc to explain why and how we're hashing the config. Otherwise I'd worry that this starts getting copy/pasted elsewhere and becomes hard to change if we ever need to

Reply:
Yeah this is a good call - Davin opened https://github.com/airbytehq/airbyte/issues/17488.